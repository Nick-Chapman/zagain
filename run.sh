#/bin/bash

# run old/new interpreter...

# new (temp: pass stdin on comman line)
stack run -- reg "$@" $(cat -); exit

# old
(niz -no-line-wrap "$@" ; echo '[no more input]') \
    | sed 's/ *  (/ (/g' \
    | sed 's/Print *"/Print "/' \
    | sed 's/Print \./Print "."/' \
    | sed 's/Print >/Print ">"/' \
    | sed 's/\\ *\\n/\\n/' \
    | cat
