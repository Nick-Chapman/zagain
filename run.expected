[release/serial: 88/840726, z-version: .z3}
(Decode 0 20229 (Call (Floc 21618) ((Con 32784) (Con 65535)) Sp))
(Decode 1 21625 (Call (Floc 21638) ((Var (Local 1))) (Local 3)))
(Decode 2 21649 (Add (Var (Global 132)) (Con 180) (Local 3)))
(Decode 3 21653 (Add (Var (Global 132)) (Var (Global 130)) (Local 4)))
(Decode 4 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 5 21661 (Sub (Var (Global 130)) (Con 6) (Global 130)))
(Decode 6 21665 (Jz (Var (Local 2)) (Branch true (Dloc 21672))))
(Decode 7 21672 (Add (Var (Global 132)) (Var (Global 130)) (Local 5)))
(Decode 8 21676 (Storew (Var (Local 5)) (Con 2) (Var (Local 1))))
(Decode 9 21681 (Return (Var (Local 5))))
(Decode 10 21631 (Storew (Var (Local 3)) (Con 1) (Var (Local 2))))
(Decode 11 21636 (Return (Var (Local 3))))
(Decode 12 20238 (Storew (Var Sp) (Con 0) (Con 1)))
(Decode 13 20243 (Call (Floc 21618) ((Con 32892) (Con 65535)) Sp))
(Decode 14 21625 (Call (Floc 21638) ((Var (Local 1))) (Local 3)))
(Decode 15 21649 (Add (Var (Global 132)) (Con 180) (Local 3)))
(Decode 16 21653 (Add (Var (Global 132)) (Var (Global 130)) (Local 4)))
(Decode 17 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 18 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 19 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 20 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 21 21697 (Jump 21657))
(Decode 22 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 23 21661 (Sub (Var (Global 130)) (Con 6) (Global 130)))
(Decode 24 21665 (Jz (Var (Local 2)) (Branch true (Dloc 21672))))
(Decode 25 21672 (Add (Var (Global 132)) (Var (Global 130)) (Local 5)))
(Decode 26 21676 (Storew (Var (Local 5)) (Con 2) (Var (Local 1))))
(Decode 27 21681 (Return (Var (Local 5))))
(Decode 28 21631 (Storew (Var (Local 3)) (Con 1) (Var (Local 2))))
(Decode 29 21636 (Return (Var (Local 3))))
(Decode 30 20252 (Call (Floc 21618) ((Con 33008) (Con 65535)) Sp))
(Decode 31 21625 (Call (Floc 21638) ((Var (Local 1))) (Local 3)))
(Decode 32 21649 (Add (Var (Global 132)) (Con 180) (Local 3)))
(Decode 33 21653 (Add (Var (Global 132)) (Var (Global 130)) (Local 4)))
(Decode 34 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 35 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 36 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 37 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 38 21697 (Jump 21657))
(Decode 39 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 40 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 41 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 42 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 43 21697 (Jump 21657))
(Decode 44 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 45 21661 (Sub (Var (Global 130)) (Con 6) (Global 130)))
(Decode 46 21665 (Jz (Var (Local 2)) (Branch true (Dloc 21672))))
(Decode 47 21672 (Add (Var (Global 132)) (Var (Global 130)) (Local 5)))
(Decode 48 21676 (Storew (Var (Local 5)) (Con 2) (Var (Local 1))))
(Decode 49 21681 (Return (Var (Local 5))))
(Decode 50 21631 (Storew (Var (Local 3)) (Con 1) (Var (Local 2))))
(Decode 51 21636 (Return (Var (Local 3))))
(Decode 52 20261 (Storew (Var Sp) (Con 0) (Con 1)))
(Decode 53 20266 (Call (Floc 21618) ((Con 28522) (Con 40)) Sp))
(Decode 54 21625 (Call (Floc 21638) ((Var (Local 1))) (Local 3)))
(Decode 55 21649 (Add (Var (Global 132)) (Con 180) (Local 3)))
(Decode 56 21653 (Add (Var (Global 132)) (Var (Global 130)) (Local 4)))
(Decode 57 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 58 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 59 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 60 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 61 21697 (Jump 21657))
(Decode 62 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 63 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 64 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 65 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 66 21697 (Jump 21657))
(Decode 67 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 68 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 69 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 70 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 71 21697 (Jump 21657))
(Decode 72 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 73 21661 (Sub (Var (Global 130)) (Con 6) (Global 130)))
(Decode 74 21665 (Jz (Var (Local 2)) (Branch true (Dloc 21672))))
(Decode 75 21672 (Add (Var (Global 132)) (Var (Global 130)) (Local 5)))
(Decode 76 21676 (Storew (Var (Local 5)) (Con 2) (Var (Local 1))))
(Decode 77 21681 (Return (Var (Local 5))))
(Decode 78 21631 (Storew (Var (Local 3)) (Con 1) (Var (Local 2))))
(Decode 79 21636 (Return (Var (Local 3))))
(Decode 80 20274 (Call (Floc 21618) ((Con 28501) (Con 200)) Sp))
(Decode 81 21625 (Call (Floc 21638) ((Var (Local 1))) (Local 3)))
(Decode 82 21649 (Add (Var (Global 132)) (Con 180) (Local 3)))
(Decode 83 21653 (Add (Var (Global 132)) (Var (Global 130)) (Local 4)))
(Decode 84 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 85 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 86 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 87 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 88 21697 (Jump 21657))
(Decode 89 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 90 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 91 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 92 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 93 21697 (Jump 21657))
(Decode 94 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 95 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 96 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 97 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 98 21697 (Jump 21657))
(Decode 99 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 100 21683 (Load_word (Var (Local 4)) (Con 2) Sp))
(Decode 101 21687 (Je ((Var Sp) (Var (Local 1))) (Branch false (Dloc 21693))))
(Decode 102 21693 (Add (Var (Local 4)) (Con 6) (Local 4)))
(Decode 103 21697 (Jump 21657))
(Decode 104 21657 (Je ((Var (Local 4)) (Var (Local 3))) (Branch false (Dloc 21683))))
(Decode 105 21661 (Sub (Var (Global 130)) (Con 6) (Global 130)))
(Decode 106 21665 (Jz (Var (Local 2)) (Branch true (Dloc 21672))))
(Decode 107 21672 (Add (Var (Global 132)) (Var (Global 130)) (Local 5)))
(Decode 108 21676 (Storew (Var (Local 5)) (Con 2) (Var (Local 1))))
(Decode 109 21681 (Return (Var (Local 5))))
(Decode 110 21631 (Storew (Var (Local 3)) (Con 1) (Var (Local 2))))
(Decode 111 21636 (Return (Var (Local 3))))
(Decode 112 20282 (Put_prop (Con 156) (Con 6) (Con 4)))
(Decode 113 20287 (Add (Var (Global 16)) (Con 2) Sp))
(Decode 114 20291 (Storew (Var (Global 10)) (Con 1) (Var Sp)))
(Decode 115 20296 (Add (Var (Global 16)) (Con 4) Sp))
(Decode 116 20300 (Storew (Var (Global 10)) (Con 2) (Var Sp)))
(Decode 117 20305 (Add (Var (Global 14)) (Con 2) Sp))
(Decode 118 20309 (Storew (Var (Global 9)) (Con 2) (Var Sp)))
